pkg/operator/validation.go

```diff
@@ -0,0 +1,178 @@
+package operator
+
+import (
+	"fmt"
+	"net"
+	"os"
+
+	"k8s.io/apimachinery/pkg/util/errors"
+)
+
+// IsValidConfig checks the validity of all configuration options.
+func IsValidConfig(cfg *Config) error {
+	errs := []error{}
+
+	errs = append(errs, IsValidNamespaceConfig(cfg))
+	errs = append(errs, IsValidListenConfig(cfg))
+	errs = append(errs, IsValidPrestoConfig(cfg))
+	errs = append(errs, IsValidHiveConfig(cfg))
+	errs = append(errs, IsValidKubeConfig(cfg.Kubeconfig))
+	errs = append(errs, IsValidPrometheusConfig(cfg))
+
+	if err := isValidTLSConfig(&cfg.APITLSConfig); err != nil {
+		errs = append(errs, fmt.Errorf("error validating apiTLSConfig: %s", err.Error()))
+	}
+
+	if err := isValidTLSConfig(&cfg.MetricsTLSConfig); err != nil {
+		errs = append(errs, fmt.Errorf("error validating metricsTLSConfig: %s", err.Error()))
+	}
+
+	if len(errs) != 0 {
+		return errors.NewAggregate(errs)
+	}
+	return nil
+}
+
+// IsValidNamespaceConfig ensures that if you are using target namespaces the all namespace field is correct.
+func IsValidNamespaceConfig(cfg *Config) error {
+	if len(cfg.TargetNamespaces) > 1 && !cfg.AllNamespaces {
+		return fmt.Errorf("must set allNamespaces if more than one namespace is passed to targetNamespaces")
+	}
+	return nil
+}
+
+// IsValidListenConfig ensures all *Listen fields are set to valid host/ports if they have a value set.
+func IsValidListenConfig(cfg *Config) error {
+	errs := []error{}
+
+	errs = append(errs, isValidHostPort(cfg.APIListen, "apiListen"))
+	errs = append(errs, isValidHostPort(cfg.MetricsListen, "metricsListen"))
+	errs = append(errs, isValidHostPort(cfg.PprofListen, "pprofListen"))
+
+	if len(errs) > 0 {
+		return errors.NewAggregate(errs)
+	}
+	return nil
+}
+
+// IsValidPrestoConfig ensure all Presto* fields are valid if provided.
+func IsValidPrestoConfig(cfg *Config) error {
+	errs := []error{}
+
+	errs = append(errs, isValidHostPort(cfg.PrestoHost, "prestoHost"))
+
+	if !cfg.PrestoUseTLS {
```

Comment:
previously we allowed config to be set even if this was set to false.  Should we continue?

Reply:
I would imagine that makes sense, i.e. rejecting a configuration where you explicitly set TLS-related flags yet disable TLS entirely for the reporting-operator -> presto communications.